src/lazy/expanded/mod.rs

```diff
@@ -321,49 +359,108 @@ impl<Encoding: Decoder, Input: IonInput> ExpandingReader<Encoding, Input> {
         Self::ref_as_ptr(evaluator)
     }
 
-    /// Updates the encoding context with the information stored in the `PendingLst`.
-    // TODO: This only works on Ion 1.0 symbol tables for now, hence the name `PendingLst`
-    fn apply_pending_lst(pending_lst: &mut PendingLst, symbol_table: &mut SymbolTable) {
+    /// Updates the encoding context with the information stored in the `PendingContextChanges`.
+    fn apply_pending_context_changes(
+        pending_changes: &mut PendingContextChanges,
+        symbol_table: &mut SymbolTable,
+        macro_table: &mut MacroTable,
+    ) {
+        if let Some(new_version) = pending_changes.switch_to_version.take() {
+            symbol_table.reset_to_version(new_version);
+            pending_changes.has_changes = false;
+            // If we're switching to a new version, the last stream item was a version marker
+            // and there are no other pending changes. The `take()` above clears the `switch_to_version`.
+            return;
+        }
+
+        if let Some(mut module) = pending_changes.take_new_active_module() {
+            std::mem::swap(symbol_table, module.symbol_table_mut());
+            std::mem::swap(macro_table, module.macro_table_mut());
+            return;
+        }
+
         // If the symbol table's `imports` field had a value of `$ion_symbol_table`, then we're
         // appending the symbols it defined to the end of our existing local symbol table.
         // Otherwise, we need to clear the existing table before appending the new symbols.
-        if !pending_lst.is_lst_append {
+        if !pending_changes.is_lst_append {
             // We're setting the symbols list, not appending to it.
             symbol_table.reset();
         }
         // `drain()` empties the pending `imported_symbols` and `symbols` lists
-        for symbol in pending_lst.imported_symbols.drain(..) {
+        for symbol in pending_changes.imported_symbols.drain(..) {
             symbol_table.add_symbol(symbol);
         }
-        for symbol in pending_lst.symbols.drain(..) {
+        for symbol in pending_changes.symbols.drain(..) {
             symbol_table.add_symbol(symbol);
         }
-        pending_lst.is_lst_append = false;
-        pending_lst.has_changes = false;
+        pending_changes.is_lst_append = false;
+        pending_changes.has_changes = false;
+    }
+
+    #[inline]
+    fn interpret_value<'top>(
+        &self,
+        value: LazyExpandedValue<'top, Encoding>,
+    ) -> IonResult<SystemStreamItem<'top, Encoding>> {
+        if value.has_annotations() && matches!(value.ion_type(), IonType::Struct | IonType::SExp) {
+            self.fully_interpret_value(value)
```

Comment:
>  Then you could interpret them in the match block around L502:

This isn't an option because `next_value` is a convenience method that most applications will not exercise. (Everything goes through `next_item`, but only some tooling and other low-level stuff uses the `next_value` wrapper.